lib/transaction/transaction.js

```diff
@@ -205,6 +205,10 @@ Transaction.prototype._isInvalidSatoshis = function() {
 };
 
 Transaction.prototype._hasFeeError = function(opts) {
+  if (this._getUnspentValue() < 0) {
```

Comment:
We can simplify this check to only run once, as `this._getUnspentValue() < 0` is the same as the `_hasMoreOutputThanInput()` check: https://github.com/dskloet/bitcore/blob/fix/transaction-error-order/lib/transaction/transaction.js#L278